mlir/lib/Gradient/Transforms/gradient_lowering.cpp

```diff
@@ -70,6 +71,8 @@ struct GradientLoweringPass : public OperationPass<ModuleOp> {
 
         // This is required to remove qubit values returned by if/for ops in the
         // quantum gradient function of the parameter-shift pattern.
+        gradientPatterns.add<JVPLoweringPattern>(gradientPatterns.getContext());
+        gradientPatterns.add<VJPLoweringPattern>(gradientPatterns.getContext());
```

Comment:
Maybe you can add these to the `populateLoweringPatterns` instead, since the comment above does not apply to these patterns.

Reply:
Done